### PR TITLE
Add `maxBefore` and `minAfter` methods to `TreeMap` and `TreeSet`

### PR DESCRIPTION
**Pull request by Psycho7**
Related to #298

### CODE REVIEW

**Comment by Psycho7** [comment]
> Do we really need both versions (the one that returns an Option and the one that doesn’t)?

In fact we don't. At first I only wanted to implement ones without `Option`. For `Map`, it's fine to return `null` when no such element, but it's bad for `Set`. So I added Option versions. In some 3rd party libraries they keep both versions (w/o option), that's why I keep them. I'd like to delete ones without `Option` just as you said.

> I think it would be nice to have these methods defined on collection.SortedSet and collection.SortedMap rather than only in the leaf classes.

Yeah this is a good idea! I even thought about adding a new trait saying they are implemented in BST. It makes sense to put them in `collection.SortedSet` and `colletion.SortedMap`. 

Thank you for your excellent suggestion!

**Comment by julienrf** [comment]
Hey @Psycho7, would you have time to address my feedback and squash your commits in the next couple of days? It would be nice to include your contribution in the next release.

**Comment by Psycho7** [comment]
Updated. The error message from Travis CI is ...

> sbt.ResolveException: download failed: com.google.guava#guava;19.0!guava.jar(bundle)

It seems like there are some network issues in Travis CI's servers. How to restart the build process? This commit should be good.

File: collections/src/main/scala/strawman/collection/SortedMap.scala [hunk 1]
```diff
@@ -59,6 +59,18 @@ trait SortedMapOps[K, +V, +CC[X, Y] <: Map[X, Y] with SortedMapOps[X, Y, CC, _],
   def firstKey: K = head._1
   def lastKey: K = last._1
 
+  /** Find the element with smallest key larger than or equal to a given key.
+    * @param key The given key.
+    * @return `None` if there is no such node.
+    */
+  def minAfter(key: K) = from(key).headOption
```
Comment (julienrf) [inline collections/src/main/scala/strawman/collection/SortedMap.scala]: Could you add the return type explicitly?

Comment (julienrf) [reply collections/src/main/scala/strawman/collection/SortedMap.scala]: Also, I’d be tempted to not have a default implementation if we always override it in leaf classes.

**Comment by Psycho7** [comment]
@julienrf Done. Previously compiling failed because `Option` was not imported, so I tried removing return types but forgot to get them back. It's awkward...

**Comment by Psycho7** [comment]
> Also, I’d be tempted to not have a default implementation if we always override it in leaf classes.

Actually that's not true. At first I also didn't want a default implementation. But there are some classes extending `SortedSet` or `SortedMap`. For example, `BitSet extends SortedSet[Int]`.

**Comment by julienrf** [comment]
OK that’s perfect, thanks for the explanation! Could you squash your commits into a single one?

**Comment by Psycho7** [comment]
@julienrf Like this?

**Comment by julienrf** [comment]
Yes, looks OK!

**Pull request merged**